packages/unmock-core/src/service/service.ts

```diff
@@ -55,58 +66,165 @@ export class Service implements IService {
     return this.matcher.matchToOperationObject(sreq);
   }
 
-  public verifyRequest(method: HTTPMethod, endpoint: string) {
-    // Throws if method + endpoint are invalid for this service
+  public updateState({
```

Comment:
Also here I would so much like if `service.updateState(input)` was just a simple call to `this.state.update(input)`, that would allow separating the concerns and produce more decoupled code.